### DISCUSSION TITLE
Feature Request: Enhanced Browser Functionality

### DISCUSSION THREAD

**Discussion by rootuseret** [zen-browser/desktop#1348] [Ideas]
## Overview
A large amount of my time, like many others, is spent in the browser. At the moment it is the main window to the digital world. The browser is the pilot's viewport, which, when properly equipped, not only performs the basic function but also helps the pilot in his work.

Today's browser products are unfortunately highly polarized. One part offers a strong minimalism, while the other part encircles the browser with “bloat-features”. At the same time, important features are often not implemented or look underdeveloped. The recent release of Firefox Nightly has given hope that not all is lost, especially the Zen browser based on it, which, in my opinion, is trying to move exactly in the philosophy of the pilot's viewport. And that's a great thing! In my search for a reliable tool, I have explored many solutions, but I stop exactly at Zen. 

Nevertheless, I find it useful to share my ideas about the chips I have highlighted from neighboring solutions.

### 1. Cloud Tab Group
Cloud tab groups are incredibly convenient to work with, as implemented in Yandex Browser. Unlike workplaces, everything is at your fingertips, and the mechanism is as simple as a Kalashnikov assault rifle. This feature allows users to place tabs in their cloud or on-premises environments. This would be useful for those who work with large amounts of information - 10-20 articles a day and want to postpone something for later. I understand what you will say — there are bookmarks for this. But, In practice, I can say that I haven't seen anyone who uses them — everyone uses either groups or just tons of open tabs. A few extra clicks to open and save bookmarks are already getting in the way, rather than helping.

### 2. Picture-in-Picture
Yandex has the best picture-in-picture player. It continues to play even when the window is closed (for example, music on YouTube). It also offers convenient settings and interaction (increased speed, quality, etc.).

### 3. Synchronized Video Translation
You can enable automatic video translation into Russian in Yandex. The audio is sent to the server, translated and returned with a voiceover. This can be implemented using the ability to connect your own translator API and the user's voice engine, for example, Google or Microsoft.

### 4. Page Translation
The built-in translation function of Yandex pages shows sentences in the original language when the cursor hovers over the translated text. This feature differs from the existing ones in that it allows users to view data without additional translation menus.

### 5. Screenshots outside browser
The Yandex screenshot works throughout the system, allowing you to take screenshots outside the browser. This integration eliminates the need for additional programs.

### 6. 3. Text extraction
You can request a summary of the video without watching it and the browser will return a response with timestamps, and that's really great! 
It would be cool to see something similar here - The ability to connect APIs such as Whisper from OpenAI to extract text or summarize information from videos.

### 7. Performance with Many Tabs
Yandex is superior to other browsers in working with a large number of tabs. Unfortunately, this also applies to Firefox. Improving your browser's performance in this area would be a huge win for science researchers.

### 1. Split Views
Edge had a split view feature similar to Terminal. It was incredibly convenient, and I'm not sure why it was deleted. It would be useful to have this feature again. And not for two, but for several windows. Just like in the terminal.

### 2. Toolkit on Sidebar
Edge's toolkit on sidebar is impressive, featuring a calculator, currency converter, translator, speed checker, and more.

### 3. Edge bar
This feature allowed the browser to collapse into a sidebar while maintaining functionality. It allowed users to open applications without the main window, which was ideal for simultaneous work with both the browser and another application on the desktop. The automatic app opening feature was convenient, especially for messengers. I don't understand why they deleted it. 

> On my own behalf, I would add to this function the possibility of a pop-up window appearing when clicking on a pinned application in the sidebar and hiding when clicking behind its area - everything is the same as in browser, only instead of the browser window itself, I will have a working application open.

### 4. Draw on screenshots
The Edge screenshot tool allows users to take notes directly on a screenshot, for example, by circling something or pointing an arrow or input some text notes. It is convenient for operational interaction with the team.

### 3. Text Voicing
Edge also allows users to choose voices and set speed for text-to-speech, which is pretty cool.

### 4. Tab Organizer
A function for organizing multiple open tabs in groups using AI in Edge. It classifies them by title value and can delete identical pages. Despite the fact that Yandex is the most convenient for organizing tabs, the integration of such a feature can help users keep track of their tabs and get rid of garbage. Also, will be helpful to sort tabs by alphabet.

### 5. Spellchecker
So far, the best in its mechanism of all that I have seen. The only thing missing is adding your own dictionary databases. Although maybe I just didn't dig deeper.

### 1. Seamless UI
Smooth transition between devices, similar to switching Spotify devices. Transferring data in one click, even between different functions such as music or browser tabs, will greatly enhance usability. Efficiency and a minimum of actions are the key points. All existing solutions are cumbersome (Sometimes just one additional click is already a serious obstacle)

### 2. API Integration
Providing the ability to connect custom APIs for features like ChatGPT, translators, or TTS-STT.

### 4. Dark Theme
Consider natively integrating a dark theme with functionalities like those found in the Dark Reader extension.

### 5. Video Speed and audio boost
Allow users to adjust video playback speeds natively more than 2x, as this is a frequently used feature. Same with audio volume.

### 7. Mobile App
We need it! :) Consider using plugins in the mobile app for convenience. This has already been done in some mobile browsers.

### 8. Fast notes
You can say that it already exists tool like sticky notes or google keep. But none of the minimalistic and good UI tools keep supports markdown. And it's very inconvenient! 

## Conclusion
I hope some of these features can be realized. Thank you for considering them!

**Comment by k-mrz** [comment]
Yandex Browser is the soulmate browser for Russian-speaking peoples and is a window to Russia for outsiders.
Maybe other Russian browser vendors or potential can make your browsing experience close to as good or equally/better than Yandex.
This browser is a one-man-dev open source project so all AI features are most surely not guaranteed but since Zen is based on Firefox, you can still enable AI using about:config flags.
You might not have noticed but the text extraction/Video-summarization is only in Yandex.Browser and no other browser.
Split view is one of Zen's features, CTRL+CLICK other tabs, right-click one of them and split the tabs.

Notes is a feature I'm def waiting for.

just check out https://www.reddit.com/r/zen_browser/ and https://zen-browser.app for more info

**Reply by rootuseret** [reply]
hello! I'm sorry if my post sounds like praising Yandex browser. This is not the case at all. as you may have noticed, I also mentioned Microsoft Edge. the problem with all commercial browsers is that they can't live without endless hanging of various useless stuff. I just described my ideas for a great product without any criticism. It is possible that the author himself or modelers will bring these functions to life. Perhaps I will join one of them myself in my free time. :)

**Comment by AntheaLiles** [comment]
Hi, 

I'd like to share some general ideas around tab groups and splits views and it seems that i'm at the right place

I'm using Zed as my daily browsers since a month now,
Switching from edge (because it remains a slow browsing experience) to Arc (that I really loved) 
then moving to Vivaldy after migrating from W11 to Linux

And finally I'm here

First, I really want to thank you for your incredible work on Zen, 
It offer a really great and modern browsing experience, really close to what Arc offers

I really love that we can hide the top bar, as I've enable the auto hide of my desktop's bar, I'm always browsing in a full screen like experience and that's awesome !

I think that the overall experience around tabs grouping and splits views can be a bit enhanced tho,

If you  can consider to allow users to manually and freely resize and reshape splited views that will be really nice
The more, I'm always a bit confused when using split views as I can't easily figure out witch tabs are actually splited

Maybe I can help to "merge" the tabs in the tab panel ?

**Reply by k-mrz** [reply]
> Hi,
> 
> I'd like to share some general ideas around tab groups and splits views and it seems that i'm at the right place
> 
> I'm using Zed as my daily browsers since a month now, Switching from edge (because it remains a slow browsing experience) to Arc (that I really loved) then moving to Vivaldy after migrating from W11 to Linux
> 
> And finally I'm here
> 
> First, I really want to thank you for your incredible work on Zen, It offer a really great and modern browsing experience, really close to what Arc offers
> 
> I really love that we can hide the top bar, as I've enable the auto hide of my desktop's bar, I'm always browsing in a full screen like experience and that's awesome !
> 
> I think that the overall experience around tabs grouping and splits views can be a bit enhanced tho,
> 
> If you can consider to allow users to manually and freely resize and reshape splited views that will be really nice The more, I'm always a bit confused when using split views as I can't easily figure out witch tabs are actually splited
I'm waiting for this too. I just wish the dev finds your request for resizing split view
> 
> Maybe I can help to "merge" the tabs in the tab panel ?
